sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -1054,5 +1056,329 @@ public virtual Response<SearchSuggestion[]> GetSearchSuggestions(string searchSt
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets time series insight types in pages synchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="Pageable{TimeSeriesType}"/> of Time Series instances types with the http response.</returns>
+        /// <seealso cref="GetTimeSeriesTypesAsync(CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        public virtual Pageable<TimeSeriesType> GetTimeSeriesTypes(CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetTimeSeriesTypes)}");
+            scope.Start();
+
+            try
+            {
+                Page<TimeSeriesType> FirstPageFunc(int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetTimeSeriesTypes)}");
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetTypesPage> getTypesResponse = _timeSeriesTypesRestClient.List(null, _clientSessionId, cancellationToken);
+                        return Page.FromValues(getTypesResponse.Value.Types, getTypesResponse.Value.ContinuationToken, getTypesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetTimeSeriesTypes)}");
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetTypesPage> getTypesResponse = _timeSeriesTypesRestClient.List(nextLink, _clientSessionId, cancellationToken);
+                        return Page.FromValues(getTypesResponse.Value.Types, getTypesResponse.Value.ContinuationToken, getTypesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series types by Time Series Types Names asynchronously.
```

Comment:
Maybe reword to 'Gets Time Series Insights types...', similar to previous methods.